instrumentation/ruby_kafka/lib/opentelemetry/instrumentation/ruby_kafka/patches/client.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module RubyKafka
+      module Patches
+        # The Client module contains the instrumentation patch the Producer#deliver_message method
+        module Client
+          def deliver_message(value, key: nil, headers: {}, topic:, partition: nil, partition_key: nil, retries: 1)
```

Comment:
This method is not actually recommended in production systems and I'm not sure its worth instrumenting, I did it to start just for the example and my own initial testing.

Reply:
fine to remove it if it isn't going to provide any value to users

Reply:
I'm going to leave it if only to serve as a simple way of demonstrating/experimenting with the instrumentation.  I also don't want to make any assumptions on how frequently the community may be actually using these features, despite the warning.